server/aws-lsp-codewhisperer/src/language-server/workspaceContextServer.ts

```diff
@@ -0,0 +1,413 @@
+import {
+    ArtifactManager,
+    FileMetadata,
+    SUPPORTED_WORKSPACE_CONTEXT_LANGUAGES,
+} from './workspaceContext/artifactManager'
+import { DEFAULT_AWS_Q_ENDPOINT_URL, DEFAULT_AWS_Q_REGION } from '../constants'
+import { WorkspaceFolderManager } from './workspaceContext/workspaceFolderManager'
+import { URI } from 'vscode-uri'
+import { getCodeWhispererLanguageIdFromPath } from './languageDetection'
+import { DependencyDiscoverer } from './workspaceContext/dependency/dependencyDiscoverer'
+
+const Q_CONTEXT_CONFIGURATION_SECTION = 'aws.q.workspaceContext'
+
+export const WorkspaceContextServer =
+    (
+        service: (
+            credentialsProvider: CredentialsProvider,
+            workspace: Workspace,
+            awsQRegion: string,
+            awsQEndpointUrl: string,
+            sdkInitializator: SDKInitializator
+        ) => CodeWhispererServiceToken
+    ): Server =>
+    features => {
+        const { logging, lsp, workspace, runtime, credentialsProvider, chat, sdkInitializator } = features
+        let workspaceFolders: WorkspaceFolder[] = []
+        let artifactManager: ArtifactManager
+        let dependencyDiscoverer: DependencyDiscoverer
+        let workspaceFolderManager: WorkspaceFolderManager
+        let isWorkflowInitialized: boolean = false
+        let isOptedIn: boolean = false
+
+        const awsQRegion = 'us-west-2' //runtime.getConfiguration('AWS_Q_REGION') ?? DEFAULT_AWS_Q_REGION
+        const awsQEndpointUrl = 'https://rts.alpha-us-west-2.codewhisperer.ai.aws.dev/' //runtime.getConfiguration('AWS_Q_ENDPOINT_URL') ?? DEFAULT_AWS_Q_ENDPOINT_URL
+        const cwsprClient = service(credentialsProvider, workspace, awsQRegion, awsQEndpointUrl, sdkInitializator)
+        lsp.addInitializer((params: InitializeParams) => {
+            workspaceFolders = params.workspaceFolders || []
+            if (params.workspaceFolders) {
+                workspaceFolders = params.workspaceFolders
+            } else {
+                logging.error(`WORKSPACE FOLDERS IS NOT SET`)
+            }
+            artifactManager = new ArtifactManager(workspace, logging, workspaceFolders)
+            dependencyDiscoverer = new DependencyDiscoverer(workspace, logging, workspaceFolders)
+            workspaceFolderManager = WorkspaceFolderManager.createInstance(
+                cwsprClient,
+                logging,
+                artifactManager,
+                workspaceFolders
+            )
+
+            return {
+                capabilities: {
+                    workspace: {
+                        workspaceFolders: {
+                            supported: true,
+                            changeNotifications: true,
+                        },
+                        fileOperations: {
+                            didCreate: {
+                                filters: [
+                                    { pattern: { glob: '**/*.{ts,js,py,java}', matches: 'file' } },
```

Comment:
the pattern should be moved to a constant file.

It should only support java and python